backend/inspirehep/tools/utils.py

```diff
@@ -77,7 +77,15 @@ def find_references(references, requested_format):
         elif len(hits) > 1:
             errors.append({"ref": ref, "line": line, "type": "ambiguous"})
         else:
-            ret.append(hits[0]["_source"][display_format])
+            source_field = hits[0]["_source"]
+            replace_string = source_field["control_number"]
+            if "texkeys" in source_field:
+                replace_string = source_field["texkeys"][0]
+            ret.append(
+                source_field[display_format].replace(
+                    "{%s" % replace_string, "{%s" % ref
+                )
+            )
```

Comment:
You can use `get` to replace the `if` and use `{{` to escape the curly bracket instead of using ugly string formatting:
```suggestion
            texkey = source_field.get("texkeys", [control_number])[0]
            ret.append(
                source_field[display_format].replace(
                    f"{{{texkey}", "{{{ref}"
                )
            )
```

Reply:
Yes, this looks better, thanks. I have to use `getattr` though since `get` doesn't work with Elasticsearch DSL's `AttrDict`